salt/states/cron.py

```diff
@@ -355,6 +355,9 @@ def present(
         return ret
 
     if special is None:
+        antidata = __salt__["cron.rm_special"](
```

Comment:
I would prefer the logic be in the `set_job` and `set_special` functions. Those functions do some logic to see what is currently set in the crontab and see if there is a difference. We should be able to add the logic in those functions to determine if there needs to be an update instead of removing it first.

Reply:
@Ch3LL I made your requested change.  And it turned out that only one of the existing modules/cron tests needed fixing.